### PR TITLE
 use Initiailizer for PyStaticMethod

### PR DESCRIPTION
**Pull request by HyunP-dev**

### CODE REVIEW

File: vm/src/builtins/staticmethod.rs [hunk 1]
```diff
@@ -62,7 +63,17 @@ impl PyStaticMethod {
     }
 }
 
-#[pyimpl(with(Callable, GetDescriptor, Constructor), flags(BASETYPE, HAS_DICT))]
+impl Initializer for PyStaticMethod {
+    type Args = PyObjectRef;
+
+    fn init(zelf: PyRef<Self>, args: Self::Args, vm: &VirtualMachine) -> PyResult<()> {
+        zelf.as_object().set_attr("__doc__", args.get_attr("__doc__", vm)?, vm)?;
+        // zelf.as_object().set_attr("__format__", args.get_attr("__format__", vm)?, vm);
```
Comment (youknowone) [inline vm/src/builtins/staticmethod.rs:71]:
```suggestion
```

**Review by youknowone — CHANGES_REQUESTED** [review]
running `cargo fmt --all` will fix the fmt failure

File: vm/src/builtins/staticmethod.rs [hunk 2]
```diff
@@ -62,7 +62,21 @@ impl PyStaticMethod {
...
 }
 
-#[pyimpl(with(Callable, GetDescriptor, Constructor), flags(BASETYPE, HAS_DICT))]
+impl Initializer for PyStaticMethod {
+    type Args = PyObjectRef;
+
+    fn init(zelf: PyRef<Self>, args: Self::Args, vm: &VirtualMachine) -> PyResult<()> {
+        zelf.as_object()
+            .set_attr("__doc__", args.get_attr("__doc__", vm)?, vm)?;
+        // zelf.as_object().set_attr("__format__", args.get_attr("__format__", vm)?, vm);
```
Comment (youknowone) [inline vm/src/builtins/staticmethod.rs:71]:
```suggestion
```

**Review by youknowone — CHANGES_REQUESTED** [review]

**Comment by Snowapril** [comment]
`expectedFailure` above the `test_decorators::test_staticmethod` need to be removed.

And it seems `test_reprlib::test_descriptors` codes are outdated and it causes the FAIL. 
You need to update `Lib/test/test_reprlib.py` up to python 3.10 (ref https://github.com/python/cpython/blob/3.10/Lib/test/test_reprlib.py#L195-L208)
Or add `expectedFailure` on that test and add separate PR for updating it.

**Comment by fanninpm** [comment]
> Or add `expectedFailure` on that test and add separate PR for updating it.

Now that #3747 has been merged, can this be rebased on a fresh copy of the `main` branch?

File: vm/src/builtins/staticmethod.rs [hunk 3]
```diff
@@ -73,7 +73,20 @@ impl PyStaticMethod {
     }
 }
 
-#[pyclass(with(Callable, GetDescriptor, Constructor), flags(BASETYPE, HAS_DICT))]
+impl Initializer for PyStaticMethod {
+    type Args = PyObjectRef;
+
+    fn init(zelf: PyRef<Self>, args: Self::Args, vm: &VirtualMachine) -> PyResult<()> {
+        zelf.as_object()
+            .set_attr("__doc__", args.get_attr("__doc__", vm)?, vm)?;
```
Comment (youknowone) [inline vm/src/builtins/staticmethod.rs:81]: this is not a fix. object is replaced, not doc.

**Review by youknowone — APPROVED** [review]
no behavior changed but harmless and preparation for next step

**Pull request merged**